src/main/scala/RemoraApp.scala

```diff
@@ -54,10 +55,28 @@ object RemoraApp extends App with nl.grons.metrics.scala.DefaultInstrumented wit
     logger.info("Reporting metricsRegistry to Cloudwatch")
     val amazonCloudWatchAsync: AmazonCloudWatchAsync = AmazonCloudWatchAsyncClientBuilder.defaultClient
 
+    // if null, cloudwatch grabs all the logs possible
+    val filterString = metricsSettings.cloudWatch.whitelist.split(" ")
+
+    val logMetricFilter = new MetricFilter() {
+      //since matches has is a string, the whitelist should be log filenames with commas with spaces
+      override def matches(string: String, metric: Metric): Boolean =  {
+        logger.debug("Metric Name:" + string)
+        if (filterString.length == 0) return true
+        for (v <- filterString) {
+            if (string.endsWith(v) || string == v ) {
+                return true
+            }
+        }
+        return false
+      }
+    }
```

Comment:
This code is duplicated in tests below, meaning that anyone can change this code here, breaking the desired behaviour, but tests would continue to pass. It would be better to pull this code out into it's own file, and implement it in a way that allows it to be used in your unit tests below

Reply:
moved the code to a utils and used that function to test